### PR TITLE
Feature user page

### PR DESCRIPTION
**Pull request by SyntaxPanda**

### CODE REVIEW

File: backend/src/test/java/de/boocom/backend/controller/UserControllerTest.java [hunk 1]
```diff
@@ -17,30 +20,56 @@
...
+                                "course":"BOJAVA231"
+                                }"""
+                        ))
+                .andExpect(status().isOk())
+                .andExpect(content().json(
+                        """
+                                {
+                                "name":"somebody",
+                                "course":"BOJAVA231"                      
+                                }"""
```
Comment (Kreyno93) [inline backend/src/test/java/de/boocom/backend/controller/UserControllerTest.java:44]: i think ure expecting ` "img" : null ` as well

File: backend/src/test/java/de/boocom/backend/controller/UserControllerTest.java [hunk 2]
```diff
@@ -17,30 +20,56 @@
...
                         "password": "123456",
                         "course":"BOJAVA231"
                         }"""
-            ))
-                .andExpect(status().isOk())
-                .andExpect(content().json(
-                        """
-                           {
-                           "name":"somebody",
-                           "course":"BOJAVA231"
```
Comment (Kreyno93) [inline backend/src/test/java/de/boocom/backend/controller/UserControllerTest.java:41]: same

File: frontend/src/App.tsx [hunk 3]
```diff
@@ -2,20 +2,20 @@ import React from 'react';
...
-function App() {
+export default function App() {
 
   return (
     <div className="App">
         <header><Header/></header>
         <Routes>
-            <Route path={"/register"} element={<RegisterPage course={""} id={""} img={""} name={""} password={""}/>}/>
+            <Route path={`/user/:id`} element={<UserPage/>}/>
+            <Route path={"/register"} element={<RegisterPage course={""} id={""} img={""} name={""} password={""} />}/>
```
Comment (Kreyno93) [inline frontend/src/App.tsx:16]: u need those props?

**Comment by Kreyno93** [comment]
please change the directory names in frontend module to lower cases

File: backend/src/test/java/de/boocom/backend/controller/UserControllerTest.java [hunk 4]
```diff
@@ -17,30 +20,56 @@
...
+        User user = mapper.readValue(content, User.class);
+
+        mockMvc.perform(MockMvcRequestBuilders.get("/api/user/" + user.getId()))
+                .andExpect(status().isOk())
+                .andExpect(content().json("""
+                        {
+                         "name":"somebody",
+                         "course":"BOJAVA231"                
+                        }"""
+                )).andExpect(jsonPath("$.id").value(user.getId()));
```
Comment (Kreyno93) [inline backend/src/test/java/de/boocom/backend/controller/UserControllerTest.java:73]: same

**Review by paffenst — APPROVED** [review]

**Pull request merged**